Remotion/Core/UnitTests/Reflection/TypeDiscovery/AssemblyLoading/ApplicationAssemblyLoaderFilterTest.cs

```diff
@@ -82,10 +82,9 @@ public void AddIgnoredAssembly ()
       Assert.That(filter.ShouldConsiderAssembly(typeof(ApplicationAssemblyLoaderFilter).Assembly.GetName()), Is.False);
     }
 
+#if NETFRAMEWORK
     [Test]
-#if !NETFRAMEWORK
     [Ignore("TODO RM-7799: Create out-of-process test infrastructure to replace tests done with app domains; RM-7808: Integrate the RoslynCodeDomProvider and renable the AssemblyCompiler tests")]
-#endif
     public void ApplicationAssemblyInclusion_DependsOnAttribute ()
     {
       string compiledAssemblyPath = Path.Combine(AppContext.BaseDirectory, "NonApplicationMarkedAssembly.dll");
```

Comment:
I'm not sure about this.

1.  you disabled the test also for .net framework, leaving the logic untested until the next task is implemented in a few weeks. That's a bit risky, although right now, there's every expectation that we will implement RM-7808 before the release.

2. Could we just switch this to assemblyloadcontext instead of appdomainrunner? Can we instantiate/execute a type in a new assemblyloadcontext in .net?

Reply:
Something went wrong here. The problem is that I cannot test that I correctly converted the test to `AssemblyLoadContext` as the `AssemblyCompiler` is not working yet. Since the `AppDomainRunner` is now NETFRAMWORK-only parts of this test need to be conditional. I suggest throwing a `NotImplementedException` and re-adding the Ignore test for .NET Core.

Reply:
Or option two removes the requirement for AppDomainRunner and solves all other issues, too? Question is, if it also solves this for .NET Framework?

Reply:
`AssemblyLoadContext` only exists in .NET Core so I can only run the test in .NET Core where the `AssemblyCompiler` is broken. Once the `AssemblyCompiler` is repaired the .NET Core test based of `AssemblyLoadContext` can be implemented. So for now I would only add a `NotImplementedException` in the .NET Core case to remind the person that enables the test again that there is something left to do.